synapse/handlers/room_member.py

```diff
@@ -1000,7 +1004,7 @@ async def _remote_join(
 
         check_complexity = self.hs.config.limit_remote_rooms.enabled
         if check_complexity and self.hs.config.limit_remote_rooms.admins_can_join:
-            check_complexity = not await self.hs.auth.is_server_admin(user)
+            check_complexity = not await self.hs.get_auth().is_server_admin(user)
```

Comment:
I think this can just be `self.auth`? Not really sure why this change was made though.

Reply:
Oh yes. I cheekily add the `hs: "HomeServer"` and "fixed" this without really thinking.

Reply:
I guess I'm not sure how this is "fixing" something?

Reply:
mypy doesn't know that `hs.auth` is a thing, as that gets magically set the first time someone calls `hs.get_auth()`, so now that it knows that `self.hs` is a `HomeServer` object it complains that it doesn't have a `auth` attribute

Reply:
Ah! We seem to store this as `self.auth` in the `__init__`, should we just use that instead and avoid the property access?

Reply:
Already done :innocent:

Reply:
This would explain why this became outdated and I didn't realize it. 🤦